### PR TITLE
Feature/add emulator support/7500

### PR DESCRIPTION
**Pull request by HemangChothani**
issue [7500]

### CODE REVIEW

File: firestore/tests/system.py [hunk 1]
```diff
@@ -43,9 +45,16 @@
 
 @pytest.fixture(scope=u"module")
 def client():
-    credentials = service_account.Credentials.from_service_account_file(FIRESTORE_CREDS)
-    project = FIRESTORE_PROJECT or credentials.project_id
-    yield firestore.Client(project=project, credentials=credentials)
+    emulator = os.getenv(FIRESTORE_EMULATOR) is not None
+    if emulator:
```
Comment (mf2199) [inline firestore/tests/system.py]: I'd make it one line: `if os.getenv(FIRESTORE_EMULATOR) is not None`

File: firestore/tests/unit/v1/test_client.py [hunk 2]
```diff
@@ -63,6 +63,22 @@ def test_constructor_explicit(self):
...
+        emulator_host = "localhost:8081"
+        with mock.patch("os.getenv") as getenv:
+            getenv.return_value = emulator_host
+            with mock.patch("grpc.insecure_channel") as factory:
+                getenv.return_value = emulator_host
+                client = self._make_one(project=self.PROJECT, credentials=credentials)
+
+        self.assertEqual(client._emulator_host, emulator_host)
+        self.assertIs(client._emulator_channel, factory.return_value)
+        factory.assert_called_once_with(emulator_host)
```
Comment (mf2199) [inline firestore/tests/unit/v1/test_client.py]: The lines 78, 79 seem to be outside of the `factory` context defined in line 73.

Comment (HemangChothani) [reply firestore/tests/unit/v1/test_client.py]: yes, but it work fine, if you want i''ll move it

File: firestore/tests/unit/v1/test_client.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def test_constructor_explicit(self):
 
     def test_constructor_with_emulator_host(self):
         from google.cloud.environment_vars import FIRESTORE_EMULATOR
-
+        import pdb;pdb.set_trace()
```
Comment (mf2199) [inline firestore/tests/unit/v1/test_client.py]: Better split this line:
`import pdb`
`pdb.set_trace()`

**Review by mf2199 — APPROVED** [review]

**Pull request closed without merging**